bpf/lib/policy.h

```diff
@@ -186,6 +196,17 @@ __policy_can_access(const void *map, struct __ctx_buff *ctx, __u32 local_id,
 				return DROP_POLICY_DENY;
 			return policy->proxy_port;
 		}
+
+		/* Check L4 any port policy */
+		key.dport = 0;
+		policy = map_lookup_elem(map, &key);
+		if (likely(policy)) {
+			account(ctx, policy);
+			*match_type = POLICY_MATCH_L4_ONLY;
+			if (unlikely(policy->deny))
+				return DROP_POLICY_DENY;
+			return CTX_ACT_OK;
```

Comment:
I'm still fuzzy on the L7 integration myself. But as this is a L4 policy, don't we need to handle that it's combined with a L7 policy and return `policy->proxy_port` here?

Reply:
Ah, yes, that's a good point.

Reply:
`policy->proxy_port` is `0`, should return `d.port` here ?

Reply:
Do you mean it's always 0? How did you check that?

Reply:
I know why `policy->proxy_port` is always 0, because when use k8s NetworkPolicy, l7 rules is empty.
But if use CiliumNetworkPolicy, when port is zero (it means all ports) we couldn't use l7 rule also.
https://github.com/cilium/cilium/blob/v1.12.1/pkg/policy/api/rule_validation.go#L390
So, in this case, we needn't consider l7 rules.

Reply:
```yaml
apiVersion: "cilium.io/v2"
kind: CiliumNetworkPolicy
metadata:
  name: "rule1"
  namespace: "cilium-test"
spec:
  endpointSelector:
    matchLabels:
      kind: echo
  ingress:
  - toPorts:
    - ports:
      - port: "0"
        protocol: TCP
      rules:
        http:
        - method: "GET"
          path: "/
```
level=warning msg="Unable to add CiliumNetworkPolicy" ciliumNetworkPolicyName=rule1 error="Invalid CiliumNetworkPolicy spec: L7 rules can not be used when a port is 0" k8sApiVersion= k8sNamespace=cilium-test subsys=k8s-watcher

Reply:
Ok. Makes sense! Thanks for double checking!